### PR TITLE
Bugfix: Passport Document followed by PoA Step Fix

### PR DESCRIPTION
**Pull request by dlewton**
# Problem

In the release candidate for version 8, there is an error where PoA steps cannot work following the upload of a passport.

# Solution

Properly check whether the upload step is a passport upload step, or not.

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [?] Has the CHANGELOG been updated?
- [n/a] Has the README been updated?
- [n/a] Has the CONTRIBUTING doc been updated?
- [n/a] Has the RELEASE_GUIDELINES been updated?
- [n/a] Has the TESTING_GUIDELINES been updated?
- [n/a] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [n/a] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [x] Have any new automated tests been implemented or the existing ones changed?
- [n/a] Have any new manual tests been written down or the existing ones changed?
- [n/a] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

File: release/BASE_32_VERSION_MAPPING.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ### Base32 version mapping to SDK version
 
 | BASE_32_VERSION | SDK_VERSION      | DATE       | NOTES                                                                                                                                                                                                                |
-| --------------- |------------------| ---------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| --------------- | ---------------- | ---------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
```
Comment (DannyvanderJagt) [inline release/BASE_32_VERSION_MAPPING.md:4]: Please don't change this file!

File: src/components/Router/HistoryRouter.tsx [hunk 2]
```diff
@@ -226,11 +226,20 @@ export default class HistoryRouter extends Component<
 
   previousStep = (): void => {
     const { step: currentStep } = this.state
-    this.setStepIndex(currentStep - 1)
+    const prevStepIndex = currentStep - 1
+
+    if (prevStepIndex < 0) {
+      // If this happens, we have a major problem
```
Comment (DannyvanderJagt) [inline src/components/Router/HistoryRouter.tsx:232]: Add error reporting here

**Review by DannyvanderJagt — APPROVED** [review]

**Review by sanchapereira — APPROVED** [review]

**Comment by it-ony** [comment]
@dlewton, thanks for adding a test case as well 👍

**Review by DannyvanderJagt — APPROVED** [review]

File: src/components/Router/CrossDeviceMobileRouter.tsx [hunk 3]
```diff
@@ -387,13 +388,23 @@ export default class CrossDeviceMobileRouter extends Component<
...
-          <HistoryRouter
-            {...this.props}
-            {...this.state}
-            crossDeviceClientError={this.setError}
-            sendClientSuccess={this.sendClientSuccess}
-            steps={steps}
-          />
+          <PoASupportedCountriesProvider
+            url={urls.onfido_api_url}
+            token={token}
```
Comment (it-ony) [inline src/components/Router/CrossDeviceMobileRouter.tsx:393]: for orchestration, I guess this won't work as the token is not set as a property. @nulrich had a similar problem with the sdksettings. Please check back with him, how he solved it.

**Review by it-ony — COMMENTED** [review]
Please check the token passed in the PoASupportedCountriesProvider. Also it comes to my mind, why do we need a token at all?

File: src/components/Router/CrossDeviceMobileRouter.tsx (hunk 3 above)
Comment (sanchapereira) [reply src/components/Router/CrossDeviceMobileRouter.tsx:393]: @it-ony the endpoint with the PoA countries + document types needs the SDK token

Comment (dlewton) [reply src/components/Router/CrossDeviceMobileRouter.tsx]: I spoke with @nulrich and made the change he suggested.

**Review by sanchapereira — APPROVED** [review]

**Review by sanchapereira — APPROVED** [review]

**Pull request merged**